File: server/models/requestModel.js
const mongoose = require('mongoose');

const requestSchema = new mongoose.Schema({
  ownerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User', 
    required: true
  },
  propertyId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Property', 
    required: true
  },
  guestId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User', 
    required: true
  },
  status: {
    type: String,
    enum: ['pending', 'accepted', 'declined'],
    default: 'pending'
  },
  bid: {
    type: Number,
    required: true
  }
}, { timestamps: true }); 

module.exports = mongoose.model('Request', requestSchema);

File: client/lib/models/request_model.dart
class RequestModel {
  String? id;
  String ownerId;
  String propertyId;
  String guestId;
  String status;
  int bid;

  RequestModel({
    this.id,
    required this.ownerId,
    required this.propertyId,
    required this.guestId,
    required this.status,
    required this.bid,
  });

  factory RequestModel.fromJson(Map<String, dynamic> json) => RequestModel(
        id: json['_id'],
        ownerId: json['ownerId'],
        propertyId: json['propertyId'],
        guestId: json['guestId'],
        status: json['status'],
        bid: json['bid'] ?? 0,
      );

  Map<String, dynamic> toJson() => {
        'id': id ?? '',
        'ownerId': ownerId,
        'propertyId': propertyId,
        'guestId': guestId,
        'status': status,
        'bid': bid,
      };
}
